Add bech32_mod for double public key encoding/decoding

gogoex opened the pull request:
This is an intermediate step to address the first 2 tasks in issue #122:
 - Create a modified version of bech32 which allows the encoding of two public keys + prefix.
 - Document it in https://github.com/navcoin/navcoin.github.io/tree/main/sources/docs

This change adds `bech32_mod::Encode` that is designed to encode 165-byte 5-bit input vector (after 8-bit to 5-bit conversion and inserting 1-byte separator, 96-byte double public key with 2-byte HRP becomes 165-byte long), and `bech32_mod::Decode` that is capable of perfectly detecting up to 5 errors with helper functions to convert betwen 8-bit based vector to 5-bit based vector and vice versa.

The process of finding a generator polynomial that is capable of the desired detection capability is discussed in `doc/bech32-mod-gen-poly` in detail.

aguycalled commented:
LGTM
I would also add support for encoding/decoding from/to `blast::DoublePublicKeys` in the class `DestinationEncoder` and the function `DecodeDestination` located in the `key_io.cpp` file. Although we can do it in a separate PR too if you prefer.

gogoex commented:
Sure. I added functions to encode/decode a double public key to key_io.cpp. Since DestinationEncoder is not exposed to outside, I added the functions directly under key_io.cpp.

mxaddict commented:
Is this PR ready for final review?
